### DISCUSSION TITLE
Vision-based planning on a F450 drone

### DISCUSSION THREAD

**Discussion by fmukwege** [ctu-mrs/mrs_uav_system#99] [Q&A]
Hello MRS team, I was planning on installing a depth camera to perform vision-based planning on a previously built f450 drone . I saw that you use a Intel realsense for your vision-based applications like SLAM. I have the Following questions : 
•	Did you use any other depth cameras than intel real sense for this drone configuration ? We plan to use a [ZED2 depth camera](https://www.stereolabs.com/zed-2/) . 
•	What codes already available in the MRS ecosystem could I run to make initial tests ? And is there some codes of the MRS ecosystem that won't work with this type of depth camera ? 
•	Is there anything I should bear in mind while using a camera with the MRS system ? 

We are currently more in an explorative phase so all advice on vision-based planning from your side is more than welcome :) !  

Best regards ,

Frank

**Comment by matemat13** [comment]
Sorry for the late answer, I didn't get any notification about your question :(

1. We tried some ZED camera long ago and didn't have great experience with it, but we don't have any hands-on experience with the ZED2. I know that some other teams use it, so I guess it's fine. Other than that, you can also check out the [Nerian](nerian.com) line of sensors with which we have good experience. It's well-made, offers a wider range of baselines (it can even work with custom-made stereo cameras) and has good configurability. But it's also more expensive than Realsense.
2. If the camera ROS driver outputs the depth images in a standard ROS format, then there should be no problem here. Regarding code - it depends on what you want to do. If you want to use it for reactive obstacle avoidance, use the [MRS bumper](https://github.com/ctu-mrs/mrs_bumper). If you want to e.g. detect other drones, you can try the [UAV detection algorithms](https://github.com/ctu-mrs/uav_detect).
3. Maybe I'd note that saving images (especially compressed) to a rosbag at a full rate of the camera can add significant CPU load, so watch out for that (this applies to ROS in general).

Best of luck!
